lib/puppet/provider/login.rb

```diff
@@ -16,30 +16,43 @@
 
 require 'json'
 
-# This method returns the information necessary in order to log in to the Oneview Appliance
-# The three possible ways of declaring the variables are, respectively:
-def login
-  options = {
+def oneview_options
+  {
     type:         'OneView',
     file_env_var: 'ONEVIEW_AUTH_FILE',
     env_var_url:  'ONEVIEW_URL',
     filename:     '/login.json'
   }
-  credentials = load_authentication_settings(options)
-  credentials[:hardware_variant] ||= 'C7000'
-  credentials
 end
 
-# This method returns the information necessary in order to log in to the Image Streamer Appliance
-# The three possible ways of declaring the variables are, respectively:
-def login_image_streamer
-  options = {
+def image_streamer_options
+  {
     type:         'Image Streamer',
     file_env_var: 'IMAGE_STREAMER_AUTH_FILE',
     env_var_url:  'IMAGE_STREAMER_URL',
     filename:     '/login_image_streamer.json'
   }
-  load_authentication_settings(options)
+end
+
+# This method returns the information necessary in order to log in to the Oneview Appliance
+# The three possible ways of declaring the variables are, respectively:
+def login
+  credentials = load_authentication_settings(oneview_options)
+  credentials[:hardware_variant] ||= 'C7000'
+  credentials
+end
+
+# This method returns the information necessary in order to log in to the Image Streamer Appliance
+# The three possible ways of declaring the variables are, respectively:
+def login_image_streamer
+  load_authentication_settings(image_streamer_options)
+end
+
+def oneview_credentials_set?
```

Comment:
> in case of changes/code refactoring related with the login, we don't need to change the ImageStreamerResource

I agree it should not be good to change the resource in case of changes to the client. 
This, however, is not being called in the client at any point. This is effectively a resource method, that is just in a separate file.